course/views.py

```diff
@@ -106,8 +106,23 @@ def get(self, request, *args, **kwargs):
         elif lti_msg:
             messages.info(request, html.escape(lti_msg))
 
-        return super().get(request, *args, **kwargs)
-
+        later_instance = None
+
+        if self.instance.is_past:
+            course = self.instance.course
+            now = timezone.now()
+            try:
+                later_instance = CourseInstance.objects.get_visible(request.user).filter(
+                    ending_time__gte=now,
+                    course=course
```

Comment:
The line is missing the comma at the end.

Reply:
In addition, as this chains multiple methods, it might be readable if every methods starts a new line. Hence, maybe:

```python
later_instance = (
    CourseInstance.objects
        .get_visible(request.user)
        .filter(course=course, ending_time__gte=now)
        .latest('starting_time')
)
```